### PR TITLE
SNOW-1344938: Add support for DataFrame.corr

### PR DESCRIPTION
**Pull request by sfc-gh-helmeleegy**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1344938

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

   Add support for DataFrame.corr.

### CODE REVIEW

**Review by sfc-gh-nkrishna — COMMENTED** [review]
Haven't taken a close look, but I'm wondering if some of this can be moved to `aggregation_utils.py` so as to be reused for groupby/resample/window functions?

**Review by sfc-gh-jkew — APPROVED** [review]
This looks pretty good. Easy to read, tests covered the issues I was thinking of.

**Comment by sfc-gh-helmeleegy** [comment]
> Haven't taken a close look, but I'm wondering if some of this can be moved to `aggregation_utils.py` so as to be reused for groupby/resample/window functions?

Yes, that's a good point. I haven't thought thoroughly about the corresponding implementation for groupby/resmaple/window. So I'd suggest to just keep the refactoring (if any) for when we start adding those other implementations.

**Comment by sfc-gh-helmeleegy** [comment]
@sfc-gh-nkrishna I think it's ready for re-review now.

File: tests/integ/modin/frame/test_aggregate.py [hunk 1]
```diff
@@ -102,6 +103,34 @@ def test_agg_basic(numeric_native_df, func, expected_union_count):
...
+        lambda x, y: np.minimum(x, y).sum().round(decimals=1),
+    ],
+)
+@sql_count_checker(query_count=0)
+def test_corr_negative(numeric_native_df, method):
+    snow_df = pd.DataFrame(numeric_native_df)
+    with pytest.raises(NotImplementedError):
+        eval_snowpark_pandas_result(
+            snow_df, numeric_native_df, lambda df: df.corr(method=method)
+        )
```
Comment (sfc-gh-nkrishna) [inline tests/integ/modin/frame/test_aggregate.py:131]:
```suggestion
        snow_df.corr(method=method)
```

**Review by sfc-gh-nkrishna — APPROVED** [review]
LGTM, just left one suggestion!

File: tests/integ/modin/frame/test_aggregate.py (hunk 1 above)
Comment (sfc-gh-helmeleegy) [reply tests/integ/modin/frame/test_aggregate.py]: Good catch!

**Pull request merged**